File: backend/spring-boot-kotlin/src/main/java/org/bootstrapbugz/api/auth/payload/request/ResetPasswordRequest.java
package org.bootstrapbugz.api.auth.payload.request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import org.bootstrapbugz.api.shared.constants.Regex;
import org.bootstrapbugz.api.shared.validator.FieldMatch;

@FieldMatch(first = "password", second = "confirmPassword", message = "{passwords.doNotMatch}")
public record ResetPasswordRequest(
    @NotBlank(message = "{token.required}") String token,
    @NotBlank(message = "{password.required}")
        @Pattern(regexp = Regex.PASSWORD, message = "{password.invalid}")
        String password,
    @NotBlank(message = "{password.required}")
        @Pattern(regexp = Regex.PASSWORD, message = "{password.invalid}")
        String confirmPassword) {}

File: backend/spring-boot-kotlin/src/main/kotlin/org/bootstrapbugz/api/auth/payload/request/ResetPasswordRequest.kt
package org.bootstrapbugz.api.auth.payload.request

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Pattern
import org.bootstrapbugz.api.shared.constants.Regex
import org.bootstrapbugz.api.shared.validator.FieldMatch

@FieldMatch(first = "password", second = "confirmPassword", message = "{passwords.doNotMatch}")
data class ResetPasswordRequest(
    @NotBlank(message = "{token.required}")
    val token: String?,
    @NotBlank(message = "{password.required}")
    @field:Pattern(
        regexp = Regex.PASSWORD,
        message = "{password.invalid}"
    )
    val password: String?,
    @NotBlank(message = "{password.required}")
    @field:Pattern(regexp = Regex.PASSWORD, message = "{password.invalid}")
    val confirmPassword: String?
)
